src/battle/main_character_attack.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include "battle.h"
+#include "../creature/main_character/main_character.h"
+
+#pragma once
+/**
+ * @brief it handels the hit and damage roll for the main character
```

Comment:
```suggestion
 * @brief it handles the hit and damage roll for the main character
```

Also, please remove the empty line between the comments and the function declaration. The empty line should be between the comments and the includes.